SingularityBase/src/main/java/com/hubspot/singularity/SingularityRequestParent.java

```diff
@@ -24,28 +24,32 @@
   private final Optional<SingularityExpiringPause> expiringPause;
   private final Optional<SingularityExpiringScale> expiringScale;
   private final Optional<SingularityExpiringSkipHealthchecks> expiringSkipHealthchecks;
+  private final Optional<Boolean> shuffleOptOut;
   private final Optional<SingularityTaskIdsByStatus> taskIds;
 
   public SingularityRequestParent(SingularityRequest request, RequestState state) {
     this(request, state, Optional.empty());
   }
 
   public SingularityRequestParent(SingularityRequest request, RequestState state, Optional<SingularityRequestDeployState> requestDeployState) {
-    this(request, state, requestDeployState, Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty());
+    this(request, state, requestDeployState, Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty());
   }
 
   @JsonCreator
-  public SingularityRequestParent(@JsonProperty("request") SingularityRequest request,
-                                  @JsonProperty("state") RequestState state,
-                                  @JsonProperty("requestDeployState") Optional<SingularityRequestDeployState> requestDeployState,
-                                  @JsonProperty("activeDeploy") Optional<SingularityDeploy> activeDeploy,
-                                  @JsonProperty("pendingDeploy") Optional<SingularityDeploy> pendingDeploy,
-                                  @JsonProperty("pendingDeployState") Optional<SingularityPendingDeploy> pendingDeployState,
-                                  @JsonProperty("expiringBounce") Optional<SingularityExpiringBounce> expiringBounce,
-                                  @JsonProperty("expiringPause") Optional<SingularityExpiringPause> expiringPause,
-                                  @JsonProperty("expiringScale") Optional<SingularityExpiringScale> expiringScale,
-                                  @JsonProperty("expiringSkipHealthchecks") Optional<SingularityExpiringSkipHealthchecks> expiringSkipHealthchecks,
-                                  @JsonProperty("taskIds") Optional<SingularityTaskIdsByStatus> taskIds) {
+  public SingularityRequestParent(
```

Comment:
Few things here:
- If we are changing this constructor, we'd need to also keep the old one (minus annotations) so that we don't break anyone's builds
- This field could likely default to `false` rather than being an Optional (we made that mistake in a few places and have tried to not do it for net new stuff)
- I'm not sure that we need to go through adding this to the request parent object. If this is just for the UI, we may want to just make a separate api call to load up the blacklist (Which should always be a small amount of data. The SingularityRequestParent object is build and used in a lot of places, some on expensive paths where speed matters. So, adding one additional zk call to every one of them could really add up. The alternative of making a small/cheap API call from the UI seems more favorable

Reply:
Yeah, it definitely felt dirty as I did it. Though I agree that the overhead isn't worth it for this feature and will change it, counterargument just in case - attaching this information to SingularityRequestParent keeps this more consistent with similar request attributes accessible through the UI, like skipping healthchecks.

For the first bullet, I assumed that this class was a wrapper Singularity put around raw SingularityRequest instances to attach its own metadata for the UI. If it's used outside this project or on critical paths, then I'm definitely switching back to the UI making direct API calls to the shuffle resource.

Reply:
Most clients just receive this object, but there are plenty of places where it could be constructed in things like unit tests. As for the first argument, many of those fields are used in things that are a bit more on the critical path for launching tasks, while shuffle is really only relevant to that single usagepoller